In [4]:
import sys
from pathlib import Path
import os
import shutil



prj_path = Path(str(Path("../../../").resolve()) + "/")

print(prj_path)


res_folder = Path(prj_path, "papers/pgm22/results/synthetic/1000/")

set_names = ["set4_aggregated", "set5"]


rfiles = [f for f in sum([[str(Path(s, f)) for f in os.listdir(Path(res_folder,s))]  for s in set_names], []) if f.endswith(".csv")]

len(rfiles)

/Users/rcabanas/GoogleDrive/IDSIA/causality/dev/credici


115

In [61]:
f = rfiles[0]
import pandas as pd



def extract_data(f):
    df = pd.read_csv(Path(res_folder, f), index_col=0)[["time_learn", "method", "selector","ps1"]].dropna()
    #df["num_runs"] = f.split("_x")[-1].split("_")[0]
    df = df.loc[(df.selector==True) & (df.method=="EMCC")][['time_learn', "ps1"]]
    df["num_runs"] = int(f.split("_x")[-1].split("_")[0])
    df["time_per_run"] = df['time_learn']/df["num_runs"]

    return df

data = pd.concat([extract_data(f) for f in rfiles])

data


time_learn    ps1  num_runs  time_per_run
2        316.0  0.000        80        3.9500
3       3968.0  0.695        80       49.6000
4       4475.0  0.773        80       55.9375
5      30689.0  0.950        80      383.6125
6      41704.0  0.962        80      521.3000
..         ...    ...       ...           ...
3    1963856.0  0.634        80    24548.2000
3    2359490.0  0.747        80    29493.6250
3    2644577.0  0.839        80    33057.2125
3    2979430.0  0.885        80    37242.8750
3    3268816.0  1.000        80    40860.2000

[1035 rows x 4 columns]

In [68]:
data.loc[data.ps1==1].time_per_run.max()/1000

0.18636024999999998

In [67]:
data.loc[data.ps1==1].time_per_run.mean()/1000

7.654242309322033

The highest time per run, considering 300 iterations is 55.9s.
The average in the cases with P(S=1)=1 is 7.65.


